src/systems/systems/src/pneumatic/mod.rs

```diff
@@ -49,11 +50,78 @@ pub trait ControllablePneumaticValve: PneumaticValve {
 }
 
 pub trait PneumaticContainer {
+    const HEAT_CAPACITY_RATIO: f64 = 1.4;
+    const HEAT_TRANSFER_SPEED: f64 = 2.5;
+    const HEAT_TRANSFER_COEFF: f64 = 1e-2;
+
     fn pressure(&self) -> Pressure;
+    // TODO: remove?
     fn volume(&self) -> Volume; // Not the volume of gas, but the physical measurements
     fn temperature(&self) -> ThermodynamicTemperature;
-    fn change_fluid_amount(&mut self, fluid_amount: Volume);
+    fn mass(&self) -> Mass;
+    fn change_fluid_amount(
+        &mut self,
+        fluid_amount: Mass,
+        fluid_temperature: ThermodynamicTemperature,
+    );
     fn update_temperature(&mut self, temperature_change: TemperatureInterval);
+
+    fn get_mass_flow_for_target_pressure(&self, target_pressure: Pressure) -> Mass {
+        ((target_pressure.get::<pascal>() / self.pressure().get::<pascal>())
+            .powf(1. / Self::HEAT_CAPACITY_RATIO)
+            - 1.)
+            * self.mass()
+    }
+
+    fn get_mass_flow_for_equilibrium(&self, other: &impl PneumaticContainer) -> Mass {
+        let p1 = self
+            .pressure()
+            .get::<pascal>()
+            .powf(1. / Self::HEAT_CAPACITY_RATIO);
+        let p2 = other
+            .pressure()
+            .get::<pascal>()
+            .powf(1. / Self::HEAT_CAPACITY_RATIO);
+        let m1 = self.mass().get::<kilogram>();
+        let m2 = other.mass().get::<kilogram>();
+        Mass::new::<kilogram>(m1 * m2 * (p2 - p1) / (m2 * p1 + m1 * p2))
+    }
+
+    /// Transfer heat between two containers depending on the temperature difference
+    fn heat_conduction(
```

Comment:
This is an interesting function. From what I see, it only exists to apply a temperature change to this container, after the change has been applied to the connected container?

I think the function is obsolete when we implement my suggestion on `heat_conduction_single`.

Reply:
Yes you are right.